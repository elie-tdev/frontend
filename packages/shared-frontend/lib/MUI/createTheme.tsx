import { createTheme as createMuiTheme } from '@mui/material/styles'

import { makeDefaultTheme } from './defaultTheme'
import { BAIThemeOptions } from './types'
import { createDefaultPaletteOptions } from './utils/palette'
import { createDefaultSpacingOptions } from './utils/spacing'

const createThemeBase = (themeOptions: BAIThemeOptions, isTest = false) => {
  createDefaultSpacingOptions(themeOptions)
  createDefaultPaletteOptions(themeOptions)
  return makeDefaultTheme(createMuiTheme(themeOptions), !isTest)
}

export const createTheme = (themeOptions: BAIThemeOptions = {}) =>
  createThemeBase(themeOptions)

export const createThemeTest = (themeOptions: BAIThemeOptions = {}) =>
  createThemeBase(themeOptions, true)
