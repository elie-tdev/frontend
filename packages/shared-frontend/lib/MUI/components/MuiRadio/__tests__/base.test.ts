import { MuiRadio } from '../'
import { createThemeTest } from '../../../createTheme'

describe('Radio', () => {
  const theme = createThemeTest()
  const component = MuiRadio(theme)

  it('should be consistent with previous versions: ', () => {
    expect(component).toMatchSnapshot()
  })
})
