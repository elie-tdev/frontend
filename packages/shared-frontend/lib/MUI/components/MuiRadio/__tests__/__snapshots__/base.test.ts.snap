// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Radio should be consistent with previous versions:  1`] = `
Object {
  "defaultProps": Object {
    "checkedIcon": <Unknown />,
    "icon": <Unknown />,
  },
  "styleOverrides": Object {
    "colorPrimary": Object {
      "& .BAIRadio-wrapper svg": Object {
        "color": "#FFFFFF",
      },
      "&.Mui-focusVisible .BAIRadio-wrapper": Object {
        "borderColor": "rgb(18, 82, 148)",
        "boxShadow": "0px 0px 0px 4px rgba(72, 145, 220, 0.5)",
      },
      "&:hover, &.Mui-checked:hover": Object {
        "backgroundColor": "transparent",
        "color": "rgb(18, 82, 148)",
      },
      "color": "#1B76D4",
    },
    "colorSecondary": Object {
      "& .BAIRadio-wrapper svg": Object {
        "color": "#FFFFFF",
      },
      "&.Mui-focusVisible .BAIRadio-wrapper": Object {
        "borderColor": "rgb(81, 81, 81)",
        "boxShadow": "0px 0px 0px 4px rgba(144, 144, 144, 0.5)",
      },
      "&:hover, &.Mui-checked:hover": Object {
        "backgroundColor": "transparent",
        "color": "rgb(81, 81, 81)",
      },
      "color": "#757575",
    },
    "root": Object {
      "& .BAIRadio-wrapper": Object {
        "backgroundColor": "transparent",
        "borderColor": "#bdbdbd",
        "borderRadius": "500px",
        "borderStyle": "solid",
        "borderWidth": 2,
        "contain": "layout paint",
        "transition": "background-color 300ms ease",
        "userSelect": "none",
      },
      "&.Mui-checked": Object {
        "& .BAIRadio-wrapper": Object {
          "background": "currentcolor",
          "borderColor": "currentcolor",
        },
      },
      "&.Mui-disabled": Object {
        "& .BAIRadio-wrapper": Object {
          "opacity": 0.3,
        },
        "color": "#bdbdbd",
      },
      "&.Mui-focusVisible": Object {
        "boxShadow": "none",
      },
      "&:hover": Object {
        "background": "none",
      },
      "color": "#9e9e9e",
    },
  },
}
`;
